backend/src/schema/resolvers/posts.js

```diff
@@ -70,5 +71,80 @@ export default {
 
       return post.properties
     },
+    AddPostEmotions: async (object, params, context, resolveInfo) => {
+      const session = context.driver.session()
+      const { to, data } = params
+      const { user } = context
+      const transactionRes = await session.run(
+        `MATCH (userFrom:User {id: $user.id}), (postTo:Post {id: $to.id})
+        MERGE (userFrom)-[emotedRelation:EMOTED {emotion: $data.emotion}]->(postTo)
+        RETURN userFrom, postTo, emotedRelation`,
+        { user, to, data },
+      )
+      session.close()
+      const [emoted] = transactionRes.records.map(record => {
+        return {
+          from: { ...record.get('userFrom').properties },
+          to: { ...record.get('postTo').properties },
+          ...record.get('emotedRelation').properties,
+        }
+      })
+      return emoted
+    },
+    RemovePostEmotions: async (object, params, context, resolveInfo) => {
+      const session = context.driver.session()
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-08-06T20:23:34Z" title="Tuesday, August 6th 2019, 10:23:34 pm +02:00">Aug 6, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -70,5 +71,76 @@ export default {
 
       return post.properties
     },
+    AddPostEmotions: async (object, params, context, resolveInfo) => {
+      const session = context.driver.session()
+      const { from, to, data } = params
+      const transactionRes = await session.run(
+        `MATCH (userFrom:User {id: $from.id}), (postTo:Post {id: $to.id})
+        MERGE (userFrom)-[emotedRelation:EMOTED {emotion: $data.emotion}]->(postTo)
+        RETURN userFrom, postTo, emotedRelation`,
+        { from, to, data },
+      )
+      session.close()
+      const [emoted] = transactionRes.records.map(record => {
+        return {
+          from: { id: record.get('userFrom').properties.id },
+          to: { id: record.get('postTo').properties.id },
+          emotion: record.get('emotedRelation').properties.emotion,
+        }
+      })
+      return emoted
+    },
+    RemovePostEmotions: async (object, params, context, resolveInfo) => {
+      const session = context.driver.session()
+      const { from, to, data } = params
```


```suggestion
      const { to, data } = params
      const { id: from } = context.user
```

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-07T05:03:24Z" title="Wednesday, August 7th 2019, 7:03:24 am +02:00">Aug 7, 2019</time>_

---


didn't think of that, but like it!!